infinitic-engine/src/main/kotlin/io/infinitic/engine/taskManager/engines/TaskEngine.kt

```diff
@@ -22,11 +22,11 @@ import io.infinitic.common.tasks.messages.interfaces.TaskAttemptMessage
 import io.infinitic.common.tasks.states.TaskEngineState
 import io.infinitic.engine.taskManager.storage.TaskStateStorage
 
-class TaskEngine(
-    val storage: TaskStateStorage,
-    val dispatcher: Dispatcher
+open class TaskEngine(
+    open val storage: TaskStateStorage,
+    open val dispatcher: Dispatcher
```

Comment:
```suggestion
    protected val storage: TaskStateStorage,
    protected val dispatcher: Dispatcher
```

If you don't change types in the subclass you don't need the `open` modifier, you only need `protected` to make them available in the subclass.